unit_tests/memory_alloc/test_db_private_alloc.cpp

```diff
@@ -252,7 +252,7 @@ namespace test_memalloc
 
     test_common::custom_assert (step == results.get_step_count ());
 
-    delete pointers;
+    delete [] pointers;
```

Comment:
While cppcheck reported some error, it was not the actuall error.
This is the error that cppcheck 1.9 reported:
```
unit_tests/memory_alloc/test_db_private_alloc.cpp:255:15: error: The deallocation of an auto-variable results in undefined behaviour. You should only free memory that has been allocated dynamically. [autovarInvalidDeallocation]
    delete [] pointers;
              ^
```

With cppcheck 2.4, the error is no longer reported.